src/main.rs

```diff
@@ -150,13 +166,26 @@ fn main() -> Result<()> {
                 LAST.with(|last| *last.borrow_mut() = now);
                 let datetime: DateTime<Utc> = now.into();
                 let time = datetime.format("%H%M%S.00");
-                let gpgga = format!(
-                    "$GPGGA,{},{:02}{:010.7},{},{:03}{:010.7},{},4,12,1.3,{:.2},M,0.0,M,1.7,0078",
-                    time, lat_deg, lat_min, lat_dir, lon_deg, lon_min, lon_dir, heightf
-                );
-                let checksum = checksum(gpgga.as_bytes());
-                let gpgga = format!("{}*{:X}\r\n", gpgga, checksum);
-                buf.write_all(gpgga.as_bytes()).unwrap();
+                let message = match &opt.area_id {
+                    Some(area_id) => {
+                        let request_counter = match &opt.request_counter {
```

Comment:
Do we need the request counter to increment or can we always send the same "counter"?

Reply:
I was under the impression that this was a cli tool that users would repeatedly invoke as per their needs, so they would be providing the relevant information for these fields.

Is this incorrect? I do not see how it is possible to calculate e.g. solution ID if so

Reply:
ntripping is not typically repeatedly invoked since it provides a constant stream of data

Reply:
I see. I will refactor and ping you.

Reply:
Revised